### ISSUE TITLE
Lights do not come on if area presence is true and night state changes

### ISSUE THREAD

**Issue by sjthespian** [jseidl/hass-magic_areas#128]
For the TV room, I have my night mode state set to be true when it is dark outside and the TV is off. However, I have notices that when we turn the TV off, while the night state will become true, the lights will not turn on until the presence has first gone false and then we trigger it again. This means that we have to sit completely still in the dark for a period of time (currently 5 mins) and then move in order to trigger the lights.

With my old automation-driven triggers, any motion after the state change would trigger the lights. It seems to me that on a stage change of night or sleep, Magic Areas should re-evaluate the state of the lights in the room and react appropriately. I.e.
* If the lights are off, area presence is true, and night becomes true, turn the lights on
* If the sleep lights are off, area presence is true, and sleep becomes true, turn the sleep lights on
* If the lights are on, area presence is true, and sleep becomes true, turn the lights off and the sleep lights on
* If the lights are on, area presence is true, and night becomes false, turn the lights off

**Comment by jseidl** [comment]
> If the lights are off, area presence is true, and night becomes true, turn the lights on
> If the lights are on, area presence is true, and night becomes false, turn the lights off

That is how it's expected to work. On my living room I have night state set to "when it's dark outside" and lights will turn on if night state comes up and the area is occupied.

> If the sleep lights are off, area presence is true, and sleep becomes true, turn the sleep lights on
> If the lights are on, area presence is true, and sleep becomes true, turn the lights off and the sleep lights on

I believe sleep lights are only consider on area state change to occupied, this is being fixed by #126 for release 3.0.0.

**Comment by sjthespian** [comment]
That is definitely not what I'm seeing. I know I was sitting in the dark last night when I turned the TV off. The lights didn't come on until I had left the room for 5 mins and came back in. Let me make sure I can reliably reproduce this, and I'll add my config and a debug log to the issue.

**Comment by jseidl** [comment]
Sure thing. You can use an `input_boolean` + `template binary sensor` to make a "manual night mode" if you wanna zero-out if the state change is indeed not triggering the lights.

**Comment by sjthespian** [comment]
I haven't had a chance to capture a debug log yet, but the behavior is pretty consistent. 

The night state in this case is triggered by a sensor, it's only true when it's dark outside (from  a luminance sensor) and we aren't watching TV (the projector screen relay is off):
```yaml
- platform: template
  sensors:
    dark_outside:
      friendly_name: Dark Outside
      value_template: "{{ states('sensor.front_sidewalk_luminance')|float < 200.0 }}"
    dark_outside_projector_off:
      friendly_name: Dark Outside and Projector Off
      value_template: >-
        {% if is_state('binary_sensor.projector_screen_trigger','on') %}
          false
        {% else %}
          {{ states('sensor.front_sidewalk_luminance')|float < 200.0 }}
        {% endif %}
```
In our case, we turn off the TV, the binary_sensor.project_screen_trigger goes off, and I can see binary_sensor.dark_outside_projector_off go true. However, the table lamp that I have set to trigger when there is motion in that area doesn't go on until the area presence first goes off. Then it will trigger. However, the hall like that triggers from the hall area goes on almost immediately -- with or without motion. I'm also noticing that that particular light triggers when the night sensor goes true the first time in the evening, again with or without motion. This may be another bug, or an undocumented feature. :)

**Comment by andyinno** [comment]
I would like to confirm that I observe the same behavior.

I have multiple motion sensors in the same room. The sensors provide also the illuminance of the room. (xiaomi motion sensors)
I have defined a template that takes the sensors illuminance of the room in order to define if the room is dark.

Problem is:
- sometimes if motion is already detected. room changes to dark, lights does not turn on.

My setup uses input_numbers for define the illuminance thresold. If I change the input_number in order to switch to bright and then to dark again, 90% times lights turn on.

Seen that while I am in the room, the sensors sends again and again the "motion detected" status (they are 3 and I am sometimes walking in the room) I suspect that some kind of re-evaluation needs to be performed.

I can confirm that switching to bright and to dark turns on the lights, therefore it needs to be some kind of bug or race condition.

**Comment by tane64** [comment]
Same when the State of sleep entity is active.

Sample:
My lights in room will turn on when the sun is below en there is movement (entity used to track area night state) Works fine

When the screens are down in this room will also be dark at daytime.
I'm setting the State of sleep to on when the lux is below 20 lx this will not work

[http://snpy.in/B6FAoM](url)

**Comment by sjthespian** [comment]
I can confirm that this is fixed in 3.0.0. I just had my monitor backlights come on when I enabled sleep mode and presence was already active in my home office.
